app/content-panel/blogs/create/page.tsx

```diff
@@ -1,15 +1,33 @@
+'use client';
+
 /* eslint-disable jsx-a11y/label-has-associated-control */
 
+import React from 'react';
+import { useSession } from 'next-auth/react';
 import styles from './createArticlePage.module.scss';
 import PublishArticleForm from '../../../../components/module-components/blog/PublishArticleForm';
 
+type UserRole = 'content_manager' | 'admin';
+
 const PostArticle = () => {
+  const { data: session, status } = useSession();
+
+  if (status === 'loading') {
+    return <div>Loading....</div>;
+  }
+
+  if (!session || !session.user.role) {
+    return <div>Access denied</div>;
+  }
```

Comment:
Also this here, we have middleware now that handles access.
Is the middleware not performing as expected or you wanted to handle something else here?